roles/basic_setup_rhel/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: "Subscription Manifest"
+  include_role:
+    name: theforeman.foreman.manifest
+
+- name: "Enable RHEL7 repository"
+  include_role:
+    name: theforeman.foreman.repositories
+  vars:
+    foreman_products:
+      - name: Red Hat Enterprise Linux Server
+        repository_sets:
+          - name: Red Hat Enterprise Linux 7 Server (RPMs)
+            basearch: x86_64
+            releasever: 7Server
+  when: enable_rhel7
+
+- name: "Enable RHEL8 repositories"
+  include_role:
+    name: theforeman.foreman.repositories
+  vars:
+    foreman_products:
+      - name: Red Hat Enterprise Linux for x86_64
+        repository_sets:
+          - name: Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)
+            releasever: 8
+          - name: Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)
+            releasever: 8
+  when: enable_rhel8
+
+- name: "Sync RHEL7 repository"
+  theforeman.foreman.repository_sync:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    organization: "{{ foreman_organization }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs }}"
+    product: Red Hat Enterprise Linux Server
+    wait_for_completion: false
```

Comment:
If we go this route we cannot say to users "Run this role and when it completes you are ready to go" and that goes against my understanding of the original intent.

Reply:
The alternative is that the role will take a very long time to run, as it first starts syncing `rhel-7-server-rpms` and waits for the task to complete, and only then starts to sync `rhel-8-for-x86_64-{baseos,appstream}-rpms`. In my lab (8cpu cores, 20gb mem) that takes about an hour.

My thinking is that documentation can inform the user that repositories have begun to sync by the time the role has finished running, and they can monitor progress at `https://katello.example.com/foreman_tasks/tasks` which is a page we want them to be familiar with regardless.

Reply:
My concern is that the repositories are unusable, registering a client is unusable until those repositories have synced. So we let users run a command designed to populate and put them in a state of ready to go when it's done but then they end up not in that state. So instead of monitoring this single command they kicked off for it to be "done". They now have to go hunt through the UI looking for tasks to monitor to let them know things are done, or user hammer to monitor the task or the API if they are not wanting to use the UI. I'd be all for this being an option to disable, but by default I think it should be enabled or, for me, it defeats the point and makes extra work for the user. 

I look at the goal of this role as:

 1) Run a single Ansible command
 2) When command completes, I can register a host and receive content

Is there a way, for this case, to parallelize the syncs? I know that for large sets of repositories we do want to be smart about it, for RHEL 7 + RHEL 8 I think Katello can handle managing syncing those optimally.